public/app/features/dashboard/state/PanelQueryRunner.ts

```diff
@@ -182,52 +175,59 @@ export class PanelQueryRunner {
       request.intervalMs = norm.intervalMs;
 
       // Check if we can reuse the already issued query
-      const active = state.getActiveRunner();
-      if (active) {
-        if (state.isSameQuery(ds, request)) {
-          // Maybe cancel if it has run too long?
-          console.log('Trying to execute query while last one has yet to complete, returning same promise');
-          return active;
-        } else {
-          state.cancel('Query Changed while running');
-        }
+      // const active = state.getActiveRunner();
+      // if (active) {
+      //   if (state.isSameQuery(ds, request)) {
+      //     // Maybe cancel if it has run too long?
+      //     console.log('Trying to execute query while last one has yet to complete, returning same promise');
+      //     return active;
+      //   } else {
+      //     state.cancel('Query Changed while running');
+      //   }
+      // }
+      //
+      // // Send a loading status event on slower queries
+      // loadingStateTimeoutId = window.setTimeout(() => {
+      //   if (state.getActiveRunner()) {
+      //     this.subject.next(this.state.validateStreamsAndGetPanelData());
+      //   }
+      // }, delayStateNotification || 500);
+      //
+      // const data = await state.execute(ds, request);
+
+      // this is not working 100% the outer subjects is marked as completed when this obserable completes, so only works
+      // for one request.
+      //
+      if (this.subscription) {
+        this.subscription.unsubscribe();
       }
 
-      // Send a loading status event on slower queries
-      loadingStateTimeoutId = window.setTimeout(() => {
-        if (state.getActiveRunner()) {
-          this.subject.next(this.state.validateStreamsAndGetPanelData());
-        }
-      }, delayStateNotification || 500);
-
-      const data = await state.execute(ds, request);
+      this.subscription = runRequest(ds, request).subscribe(this.subject);
```

Comment:
`.subscribe(this.subject)` - how does this work? Is the subject notified when the runRequest returns, and hence all of the subject's subscriptions are notified as a consequence?

Reply:
you can pass a subject to a subscribe function, was in the docs: https://rxjs-dev.firebaseapp.com/guide/subject 

the object you pass to subscribe should have a next, error, and complete function (Ie an observer). And every subject is an observer (according to the docs).